### ISSUE TITLE
Create a new "featured" view

### ISSUE THREAD

**Issue by ryelle** [WordPress/pattern-directory#542]
As mentioned in https://github.com/WordPress/gutenberg/issues/44501, there should be a "featured" tag for the pattern directory. Currently we have a "featured" category, which could work for this, or we could take the opportunity to switch this over to an internal keyword (since it's not _really_ meant to be used in the same way as the other categories), as you should be able to filter featured by category.

- [x] Create a new keyword term "featured", and apply the new term to patterns by wordpressdotorg
- [ ] Add a “featured” toggle to the filter navigation (needs design)
- [ ] Support showing only featured posts in categories, homepage, etc

Tech note— currently, we've done a lot of permalink rewriting to allow complex filtering of patterns with "pretty" urls. Adding in this filter will make it even more complex, so I would stick with a query parameter instead of trying to make the URLs pretty (ex, leave it as `/patterns/categories/header/?featured=true` rather than trying to inject `/featured/` in there somewhere).

@jasmussen I've added this issue separately as a "needs design", but you can combine it with the design for #395 if that's easier. There are two other design/experience questions here though: Should the front page show all patterns or only featured by default? Is there a use case for showing just “community”, or should it be “All | Featured”?

**Comment by jasmussen** [comment]
> Should the front page show all patterns or only featured by default? Is there a use case for showing just “community”, or should it be “All | Featured”?

Featured from the core set sounds right to me, but I'd love thoughts from @WordPress/gutenberg-design

**Comment by jameskoster** [comment]
I agree it makes sense to give increased prominence to the featured patterns – that seems to be the whole point of that classification existing.

That said, displaying 'all' pre-filtered by 'featured' as a default configuration might be unexpected? It's a bit unusual to have filters pre-selected, normally you have to intentionally add them.

An alternative to consider would be including 'Featured' as an ordering option:

<img width="1016" alt="Screenshot 2022-11-23 at 12 17 31" src="https://user-images.githubusercontent.com/846565/203544957-4efb3ae2-2611-4f04-b7f7-fec50f806f00.png">

It's a bit strange to order by a binary value (we'd need secondary ordering – chronological?), but at least this way it's not a toggle – you browse _all_ patterns, just with featured ones appearing first.

**Comment by adamwoodnz** [comment]
@jameskoster @jasmussen thanks for your input, so is this alternative option what we want to go with? And should that then be the default ordering on page load?

**Comment by jameskoster** [comment]
Yes it would be the default. 

That said it's just one idea. I'd still like to hear more thoughts from @WordPress/gutenberg-design.

**Comment by jasmussen** [comment]
> That said, displaying 'all' pre-filtered by 'featured' as a default configuration might be unexpected? It's a bit unusual to have filters pre-selected, normally you have to intentionally add them.

I think it might be useful to zoom out our rewind and look at the main reason for introducing a classification between "featured" and "community", and the main reason is to have a useful classification of patterns to show in the core modal pattern explorer, this one: https://github.com/WordPress/gutenberg/issues/44611 

How we then filter the homepage of the pattern directory, I think we can experiment and see what ultimately works best. Is it defaulting to featured, defaulting to community, or showing "all" intermixed? If we're open to revisiting, I personally think we can pick either of these options as a starting point.

**Comment by jameskoster** [comment]
Good point. I notice we don't have an 'All' category in the plugin. Do we need one in the directory? If we remove that category, we can keep the 'Featured' one and just make it the default selection.

**Comment by jasmussen** [comment]
I think the plugin renders patterns differently than the directory, making "all" a performance challenge. Not sure 🤔

**Comment by ryelle** [comment]
Quick update here: A new term, `featured` (59) has been added as a `wporg-pattern-keyword`, and can be requested via the API using `http://api.wordpress.org/patterns/1.0/?locale=en_US&wp-version=6.1&pattern-keywords=59`.

There are currently 126 featured patterns - all the patterns by wordpressdotorg. Right now, there is no way to see this list on the Pattern Directory website.

**Comment by richtabor** [comment]
> As mentioned in https://github.com/WordPress/gutenberg/issues/44501, there should be a "featured" tag for the pattern directory.

If tagged "Featured" these wouldn't show up in the Block Inserter as featured patterns, right?

**Comment by ryelle** [comment]
As of right now, we have 3 things going on here:

| State | Shows up in Block Inserter | View on wp.org/patterns |
|---|---|---|
| internal keyword = core | ✅ WP 5.8+ | ❌ |
| category = featured | ✅ WP 5.9+ | ✅ [archive](https://wordpress.org/patterns/categories/featured/) |
| internal keyword = featured | ❌ | ❌ |

Conceptually anything tagged "Featured" (however we implement that) should show up in the Block Inserter.

This new featured view was never implemented on wp.org/patterns because we were waiting on a design for the new nav bar. And because it wasn't implemented there, it didn't carry over into GB or core.

**Comment by richtabor** [comment]
> Conceptually anything tagged "Featured" (however we implement that) should show up in the Block Inserter.

Will these always show up in the "Featured" category within the Block Inserter? Or in their respective categories?

**Comment by ryelle** [comment]
Depends on the implementation (a new "Pattern Explorer" was started here https://github.com/WordPress/gutenberg/pull/45237). Right now they show up in both, so [this "Image and quote on a background" pattern](https://wordpress.org/patterns/pattern/image-and-quote-on-a-background/) shows up in the Featured category and the Text category.

There was [some discussion](https://github.com/WordPress/gutenberg/pull/46144#issuecomment-1332398385) about removing the Featured category (since it's not really the same as Posts, Text, etc), but AFAIK that was never picked up again.

**Comment by richtabor** [comment]
What do you think about adding a top-level filter between "Curated" and "Community" patterns? "Featured" is already in use otherwise within the Pattern Inserter. 

This is a simple UI update we could use to get it there (and also account for [additional categories](https://github.com/WordPress/gutenberg/issues/47847)): 

<img width="1357" alt="CleanShot 2023-03-28 at 12 04 08" src="https://user-images.githubusercontent.com/1813435/228300380-29264d1c-ac56-4bf9-97fb-6508fedb8cc5.png">

**Comment by annezazu** [comment]
I like this approach. I think we can start with Curated and iterate as we need to as well if it doesn't land well. I weirdly like having two words starting with C too between "Curated" and "Community".

**Comment by richtabor** [comment]
> Should the front page show all patterns or only featured by default? Is there a use case for showing just “community”, or should it be “All | Featured”?
> Featured from the core set sounds right to me, but I'd love thoughts...

Agreed. It should start with the curated set visible first.

**Comment by richtabor** [comment]
> What do you think about adding a top-level filter between "Curated" and "Community" patterns?

@mtias what do you think of this nomenclature?

**Comment by richtabor** [comment]
We should also consider moving "Wireframe" as a third top-level category, alongside "Curated" and "Community". That way we can have Wireframe Headers, Wireframe Banners, etc.

**Comment by beafialho** [comment]
I like it "Curated". I'm just wondering what would show in the dropdown if we click "Curated" and "Community"?

**Comment by jasmussen** [comment]
Noting that the term "Canonical" is being suggested for themes and plugin directories. Not sure if that's an ideal term for patterns compared to "Curated", but conceptually there's some overlap.

**Comment by richtabor** [comment]
> I'm just wondering what would show in the dropdown if we click "Curated" and "Community"?

I was thinking just keeping it a simple select, as it is today, to be the lowest effort: 

<img width="227" alt="CleanShot 2023-04-27 at 12 13 37" src="https://user-images.githubusercontent.com/1813435/234832963-b809a12f-f0dd-4fda-92e4-334c65d4f6e1.png">

**Comment by ryelle** [comment]
Closing in favor of #580, which has a new UI & more discussion.
